src/main/scala/com/fulcrumgenomics/util/GeneAnnotations.scala

```diff
@@ -25,32 +25,92 @@
 
 package com.fulcrumgenomics.util
 
-import htsjdk.samtools.util.{CoordMath, Interval}
+import htsjdk.samtools.util.{CoordMath, Locatable}
+import com.fulcrumgenomics.FgBioDef._
 
 /** Stores classes useful for storing annotation information for genes and their transcripts and exons. */
 object GeneAnnotations {
+  /**
+    * A gene with a collection of gene loci.
+    */
+  case class Gene(name: String, loci: Seq[GeneLocus]) extends Iterable[Transcript] {
+    def iterator: Iterator[Transcript] = this.loci.iterator.flatten
 
-  /** A gene with zero or more transcripts, that all are on the same transcription strand. */
-  case class Gene(contig: String, start: Int, end: Int, negativeStrand: Boolean, name: String, transcripts: Seq[Transcript])
-    extends Interval(contig, start, end, negativeStrand, name) with Iterable[Transcript] {
-    def iterator: Iterator[Transcript] = transcripts.iterator
+    /** Returns the locus with the maximum transcript mappings. */
+    def primaryLocus: GeneLocus = loci.maxBy(_.size)
   }
 
-  /** A transcript associated with a given gene (which stores the transcription strand).  Contains zero or more exons.
-    * The exons should be given in transcript order. */
-  case class Transcript(name: String, start: Int, end: Int, cdsStart: Int, cdsEnd: Int, exons: Seq[Exon]) {
+
+  /**
+    * A gene locus representing a set of transcripts that are mapped to the same location on the genome.
+    *
+    * @param transcripts the set of transcripts at the locus
+    */
+  case class GeneLocus(transcripts: Seq[Transcript]) extends Locatable with Iterable[Transcript] {
+    require(transcripts.nonEmpty, "GeneLocus cannot be generated from an empty set of transcripts.")
+    require(transcripts.forall(t => t.chrom == transcripts.head.chrom))
+    require(transcripts.forall(t => t.negativeStrand == transcripts.head.negativeStrand))
+
+    val chrom: String = transcripts.head.chrom
+    val start: Int = transcripts.minBy(_.start).start
+    val end:   Int = transcripts.maxBy(_.end).end
+    val negativeStrand: Boolean = transcripts.head.negativeStrand
+    def positiveStrand: Boolean = !negativeStrand
+
+    // Locatable implementation
+    override def getContig: String = chrom
+    override def getStart: Int = start
+    override def getEnd: Int = end
+
+    override def iterator: Iterator[Transcript] = this.transcripts.iterator
+  }
+
+
+  /**
+    * A transcript [mapping] associated with a given gene locus.  Contains zero or more exons.
+    * The exons should be given in the order of transcription.
+    */
+  case class Transcript(name: String,
+                        chrom: String,
```

Comment:
Note: I pushed `chrom` and `negativeStrand` down here for a few reasons:
1. It's just convenient to have this info on transcript
2. Most sources of annotation are per transcript mapping, so I feel like it makes sense to have all the details here and then also roll it up higher where we need it
3. It allows `Transcript` to implement Locatable which seems useful